R/utils.R

```diff
@@ -85,3 +85,26 @@ paste_line <- function (...) {
 has_dim <- function(x) {
   !is.null(attr(x, "dim"))
 }
+
+maybe <- function(value = NULL, error = NULL) {
```

Comment:
This feels more like a Rust Result than a Haskell Maybe

Reply:
Can we mark this clearly as experimental? I don't think we should start using it widely, even internally, until we're a bit more confident in the pattern.

Reply:
Good point, a Haskell maybe is just an option type with `NULL` as fallback value.

I'll mark it more clearly as experimental, I'm not sure I like this pattern either, compared to throwing and catching. It does feel boilerplaty. I guess try-catch is boilerplaty as well though.

Reply:
I'll rename in a future PR to avoid conflicts with other work.